library/kani/src/futures.rs

```diff
@@ -41,3 +44,216 @@ const NOOP_RAW_WAKER: RawWaker = {
 
     RawWaker::new(std::ptr::null(), &RawWakerVTable::new(clone_waker, noop, noop, noop))
 };
+
+static mut EXECUTOR: Scheduler = Scheduler::new();
+const MAX_TASKS: usize = 16;
```

Comment:
Can we make this a parameter of the Scheduler?

Reply:
I guess we could, but then it would have to be an argument of `spawnable_block_on` as well. Given that anything with more nondeterminism than 16 tasks is unlikely to terminate in a reasonable time frame anyway, I don't think I see the point. Could you elaborate?

Reply:
Sure. This can actually affect the result of the verification, and none of our APIs are reflecting that.

Quick questions to help me understand this:
1- What will happen if by any chance we hit the MAX?
2- Does using a lower MAX help with the verification time?

Reply:
If lowering may improve the verification time, wouldn't it be helpful for users to be able to set this value?

Reply:
Well, as soon as the `Vec` performance improvements land, I'd like to replace this by a vector anyway. This is more flexible and does not require a constant upper bound. Therefore, I'd rather not expose the maximum number of tasks in the API.

Reply:
Now using `Vec` instead of an array, so this is resolved.